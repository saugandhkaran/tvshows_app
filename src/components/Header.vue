<template>
  <nav class="flex-container">
   <router-link to="/" exact><h2>PlayTV24x7</h2></router-link>
    <section class="flex-container menu-items">
      <router-link to="/browse"><h3>Browse</h3></router-link>
      <router-link to="/search"><h3>Search</h3></router-link>
    </section>
  </nav>
</template>

<script>
export default {
  name: 'Header',
  props: {
    showObject: Object,
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style lang="scss" scoped>
nav {
  position: fixed;
  width: 100%;
  height: 73px;
  background-color: rgb(18,21,26)
}
.flex-container {
  padding: 0px 20px;
  align-items: center;
  color: rgb(138, 6, 98);
  h2 {
    color: rgb(138, 6, 98);
  }
  .menu-items {
    margin-right: 20px;
  }
  a {
    margin-left: 30px;
  }
}

</style>
