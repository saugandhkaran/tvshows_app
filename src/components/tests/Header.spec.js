import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import Header from '../Header';

describe('header', () => {
  
  it('should match the header html', () => {
    const wrapper = shallowMount(Header, {
      stubs: {
        RouterLink: RouterLinkStub,
      },
    })
    expect(wrapper.element).toMatchSnapshot();
  });
  
});
