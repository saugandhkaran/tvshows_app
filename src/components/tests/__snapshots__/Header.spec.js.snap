// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header should match the header html 1`] = `
<nav
  class="flex-container"
>
  <a>
    <h2>
      PlayTV24x7
    </h2>
  </a>
   
  <section
    class="flex-container menu-items"
  >
    <a>
      <h3>
        Browse
      </h3>
    </a>
     
    <a>
      <h3>
        Search
      </h3>
    </a>
  </section>
</nav>
`;
